docs/adrs/006-airgap-rpm-support.md

```diff
@@ -0,0 +1,55 @@
+# RPM Airgap Support
+
+We should better support RPM installation in air gapped environments
+
+## Established
+
+2023-10-20
+
+## Revisit by
+
+2024-10-01
+
+## Subject
+
+1. When releasing, we bundle RPMs into groups by OS, copy them to a directory, use createrepo to generate local repo information, tarball the directory, and add it as an artifact on the release.
+2. Given install method 'rpm' and variable 'INSTALL_RKE2_ARTIFACT_PATH', when run, the install.sh script looks for a local rpm installation at the given path and installs rke2 using that repo.
```

Comment:
Does this work? I didn't even know that you could point yum at a local repo-dir instead of a http/https server.

What are the advantages of shipping a full snapshot of repo files, over just shipping the 4 RPMs (rke2-common, rke2-agent, rke2-server, and rke2-selinux)?

Reply:
Our selinux policies require rpm installation, right?
The use case is to enable air gapped installation on selinux enforcing environments.

Local yum repositories are possible: https://rpmdeb.com/devops-articles/how-to-create-local-yum-repository/

Reply:
createrepo is not a core linux utility, and is only necessary to generate the rpm repo files, running this and bundling it up for the user enables a much simpler approach, just like bundling our other images into an archive. 

Another way to think of this approach is just like the tar installation, but for rpms.

Reply:
Thank you for the context on rancher-system-agent and system-agent-installer-rke2, I will add that to the context part of this doc! 
It appears rancher-system-agent is just a program that watches for plans from something else, it would be unaffected by this change.

The system-agent-installer-rke2 appears to be a plan for the rancher-system-agent which downloads the install script and runs it with install method = 'tar'. I am not sure how this affects rpm installs, but since the RPMs install the binaries in similar places as the docs for the manual install, I would assume this would still work. I am sure that we test this upgrade path already and that the directory forces the proper selinux tags on the binaries, the change proposed in this ADR won't affect that at all.

Reply:
I have found several tools which bundle all dependencies for rke2 so that it can be copied and installed locally on a server with no internet, but most have their own installation method. I think it would be better if we provided support for this use case. We supply almost all our dependencies currently in various repos, we just need to bundle them up by operating system and enable our installer to use them.
This ADR targets the installer changes.